tensorflow/core/kernels/data/captured_function.cc

```diff
@@ -840,7 +848,24 @@ Status InstantiatedCapturedFunction::Run(IteratorContext* ctx,
             "InstantiatedCapturedFunction::Run#id=", f_opts.step_id, "#");
       },
       profiler::TraceMeLevel::kInfo);
-  TF_RETURN_IF_ERROR(lib_->RunSync(std::move(f_opts), f_handle_, &frame));
+  if (collect_usage) {
+    // Resource usage is for function execution is gathered from the executor.
+    node->record_stop(EnvTime::NowNanos());
+    TF_RETURN_IF_ERROR(lib_->RunSync(std::move(f_opts), f_handle_, &frame));
+    node->record_start(EnvTime::NowNanos());
+    if (ctx->stats_aggregator()) {
+      string prefix_with_func_name =
+          strings::StrCat(node->name(), stats_utils::kDelimiter,
+                          captured_func_->func().name());
+      ctx->stats_aggregator()->AddToHistogram(
+          stats_utils::ExecutionTimeHistogramName(prefix_with_func_name),
+          {static_cast<float>(stats_collector->processing_time())},
+          node->num_elements());
+    }
```

Comment:
For consistency with `RunAsync`, this part should be taken out of the `if (collect_usage)` body.

I think it should be:

```
if (node) {
  if (collect_usage) node->record_stop(...);
  TF_RETURN_IF_ERROR(lib_->RunSync(std::move(f_opts), f_handle_, &frame));
  if (ctx->stats_aggregator()) {
    ...
  }
  node->add_processing_time(stats_collector->processing_time());
  if (collect_usage) node->record_start(...);
}
```